### PR TITLE
Ticket 6: get all comments by article_id

### PR DESCRIPTION
**Pull request by omobayed**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -110,4 +110,33 @@ describe("GET /api/articles", () => {
...
+                })
+            })
+    })
+    test("400: Error - should return bad request when passing invalid article_id", () => {
+        return request(app)
+            .get('/api/articles/car/comments')
+            .expect(400)
+            .then(({ body }) => {
+                expect(body).toEqual({ msg: "Bad Request" });
+            })
```
Comment (EddTheDuck) [inline __tests__/app.test.js:141]: Decent tests but you need to check for a 404 - if it's a valid ID that doesn't exist

Comment (EddTheDuck) [inline __tests__/app.test.js:141]: You should also test for a valid ID that exxists but has no comments. This should return an empty array.

File: models/comments.model.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const db = require("../db/connection")
+
+exports.selectCommentsByArticleId = (articleId) => {
+    return db
+        .query('SELECT * FROM comments WHERE article_id = $1;', [articleId])
+        .then((result) => {
+            // if (result.rows.length === 0) {
+            //     return Promise.reject({ status: 404, msg: "Not found" })
```
Comment (EddTheDuck) [inline models/comments.model.js]: always remove commented out code before submitting a PR

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
Pretty good but needs more tests before proceeding.

File: models/comments.model.js (hunk 2 above)
Comment (omobayed) [reply models/comments.model.js]: Removed

File: __tests__/app.test.js (hunk 1 above)
Comment (omobayed) [reply __tests__/app.test.js:141]: Added a test for this case

Comment (omobayed) [reply __tests__/app.test.js:140]: done

File: __tests__/app.test.js [hunk 3]
```diff
@@ -110,4 +110,52 @@ describe("GET /api/articles", () => {
...
+})
+
+describe("/api/articles/:article_id/comments", () => {
+    test("200: should respond with an array of comments for the given article_id ", () => {
+        return request(app)
+            .get("/api/articles/1/comments")
+            .expect(200)
+            .then(({ body }) => {
+                const { comments } = body;
+                expect(comments).toBeInstanceOf(Array);
```
Comment (NRMSMITH) [inline __tests__/app.test.js:122]: I would like to see a check for the order :) (most recent first)

File: __tests__/app.test.js (hunk 1 above)
Comment (omobayed) [reply __tests__/app.test.js:141]: I was working on this already, but wasn't sure if it's needed to be done in this ticket or not

**Review by NRMSMITH — COMMENTED** [review]
Make sure you put an expect to check that the order is in most recent comments first! Resubmit once you've done 😄

File: __tests__/app.test.js (hunk 3 above)
Comment (omobayed) [reply __tests__/app.test.js:122]: sorry I missed that

**Review by NRMSMITH — APPROVED** [review]
🚀 All good to go!

**Pull request merged**